scripts/install-helpers/baremetal-coco/install.sh

```diff
@@ -0,0 +1,733 @@
+        exit 1
+
+    # Add the new secret to the existing secret
+    jq --argjson data "$PULL_SECRET_JSON" '.auths |= ($data + .)' cluster-pull-secret.json >cluster-pull-secret-mod.json || exit 1
+
+    # Set the image pull secret
+    oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=cluster-pull-secret-mod.json || exit 1
+
+}
+
+#Function to deploy the OpenShift sandboxed containers (OSC) operator
+function deploy_osc_operator() {
+    echo "OpenShift sandboxed containers operator | starting the deployment"
+    # Apply the operator manifests
+    apply_operator_manifests
+
+    wait_for_deployment controller-manager openshift-sandboxed-containers-operator || return 1
+
+    # Wait for the service endpoints IP to be available
+    wait_for_service_ep_ip webhook-service openshift-sandboxed-containers-operator || return 1
+
+    echo "OpenShift sandboxed containers operator | deployment finished successfully"
+}
+
+# Function to deploy NodeFeatureDiscovery (NFD) operator
+function deploy_nfd_operator() {
+    echo "Node Feature Discovery operator | starting the deployment"
+
+    pushd nfd || return 1
+    oc apply -f ns.yaml || return 1
+    oc apply -f og.yaml || return 1
+    oc apply -f subs.yaml || return 1
+    popd || return 1
+
+    wait_for_deployment nfd-controller-manager openshift-nfd || return 1
+    echo "Node Feature Discovery operator | deployment finished successfully"
+}
+
+function create_intel_node_feature_rules() {
+    echo "Node Feature Discovery operator | creating intel node feature rules"
+
+    pushd nfd || return 1
+    oc apply -f intel-rules.yaml || return 1
+    popd || return 1
+
+    echo "Node Feature Discovery operator | node feature rules successfully created"
+}
+
+function deploy_intel_device_plugins() {
+    echo "Intel Device Plugins operator | starting the deployment"
+
+    pushd intel-dpo || return 1
+    oc apply -f install_operator.yaml || return 1
+    wait_for_deployment inteldeviceplugins-controller-manager openshift-operators || return 1
+    oc apply -f sgx_device_plugin.yaml || return 1
+    popd || return 1
+
+    echo "Intel Device Plugins operator | deployment finished successfully"
+}
+
+function create_amd_node_feature_rules() {
+    echo "Node Feature Discovery operator | creating amd node feature rules"
+
+    pushd nfd || return 1
+    oc apply -f amd-rules.yaml || return 1
+    popd || return 1
+
+    echo "Node Feature Discovery operator | node feature rules successfully created"
+}
+
+# Function to create KataConfig based on TEE type
+function create_kataconfig() {
+    local tee_type=${1}
+    local label='coco: "true"'
+
+    case $tee_type in
+    tdx)
+        label='intel.feature.node.kubernetes.io/tdx: "true"'
+        ;;
+    snp)
+        label='amd.feature.node.kubernetes.io/snp: "true"'
+        ;;
+    esac
+
+    # Create KataConfig object
+    oc apply -f - <<EOF
```

Comment:
If you do :
```suggestion
    oc apply -f - || return 1 <<EOF
```
you can...